### PR TITLE
chore: bump eclipse platform dependencies to the latest versions

### PR DESCRIPTION
**Pull request by mcollovati**

### CODE REVIEW

File: flow-client/pom.xml [hunk 1]
```diff
@@ -132,7 +132,7 @@
         <dependency>
             <groupId>org.eclipse.platform</groupId>
             <artifactId>org.eclipse.core.resources</artifactId>
-            <version>3.17.0</version>
+            <version>3.21.0</version>
             <scope>test</scope>
             <exclusions>
                 <exclusion>
```
[the remark this answers is not in the document]
Comment (mcollovati) [reply flow-client/pom.xml:138]: The issue mentioned in the commit that introduced the exclusion has been fixed.
I think we can remove the exclusion

File: flow-client/pom.xml [hunk 2]
```diff
@@ -381,6 +381,13 @@
             </plugin>
         </plugins>
     </build>
+    <repositories>
+        <repository>
+            <id>maven_central</id>
+            <name>Maven Central</name>
+            <url>https://repo.maven.apache.org/maven2/</url>
```
[the remark this answers is not in the document]
Comment (mcollovati) [reply flow-client/pom.xml:388]: Nope, added accidentally

File: flow-client/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mcollovati) [reply flow-client/pom.xml]: done

File: flow-client/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mcollovati) [reply flow-client/pom.xml]: removed

**Review by ZheSun88 — APPROVED** [review]

**Pull request merged**